internal/mempool/mempool.go

```diff
@@ -1112,7 +1239,8 @@ func (mp *TxPool) findTx(txHash *chainhash.Hash) *mining.TxDesc {
 func (g *txDescGraph) addChild(tx *mining.TxDesc, child *mining.TxDesc) {
 	txHash := *tx.Tx.Hash()
 	if _, exists := g.childrenOf[txHash]; !exists {
-		g.childrenOf[txHash] = make(map[chainhash.Hash]*mining.TxDesc)
+		g.childrenOf[txHash] = make(map[chainhash.Hash]*mining.TxDesc,
+			len(tx.Tx.MsgTx().TxOut))
```

Comment:
This can be the lesser of `len(tx.Tx.MsgTx().TxOut)` and `ancestorTrackingLimit`.